.github/workflows/test.yml

```diff
@@ -174,3 +175,76 @@ jobs:
           cp functionalAccounts.json.test functionalAccounts.json
           docker-compose up --build -d
           npm run test:api
+
+  generate_and_test_openapi_client:
+    name: Generate and test OpenAPI Client libraries
+    needs: [install-and-cache]
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout your code
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Restore node_modules
+        id: cached-node-modules
+        uses: actions/cache@v4
+        with:
+          path: node_modules
+          key: node-modules-${{ hashFiles('package-lock.json') }}-${{ env.NODE_VERSION }}
+
+      # Generate OpenAPI definition
+      - name: Start Backend for later tests and generate openapi.json
+        env:
+          ELASTICSEARCH_ENABLED: false
+          MONGODB_URI: mongodb://localhost:27017/scicat
+          EXPRESS_SESSION_SECRET: a_scicat_secret
+          JWT_SECRET: a_scicat_secret
+          PORT: 3000
+          HTTP_MAX_REDIRECTS: 5
+          HTTP_TIMEOUT: 5000
+          JWT_EXPIRES_IN: 3600
+          LOGGING_DISABLED: true
+          SITE: PSI
+          PID_PREFIX: PID.SAMPLE.PREFIX
+          DOI_PREFIX: DOI.SAMPLE.PREFIX
+          METADATA_KEYS_RETURN_LIMIT: 100
+          METADATA_PARENT_INSTANCES_RETURN_LIMIT: 100
+          ADMIN_GROUPS: ingestor
+          DELETE_GROUPS: ingestor
+          CREATE_DATASET_GROUPS: ingestor
+          CREATE_DATASET_WITH_PID_GROUPS: ingestor
+          CREATE_DATASET_PRIVILEGED_GROUPS: ingestor
+          ACCESS_GROUPS_STATIC_VALUES: "ess"
+          PROPOSAL_GROUPS: ingestor
+          SAMPLE_GROUPS: ""
+          DATASET_CREATION_VALIDATION_ENABLED: false
+          MONGODB_COLLECTION: Dataset
+          ES_PASSWORD: duo-password
+          STACK_VERSION: 8.8.2
+          CLUSTER_NAME: es-cluster
+          MEM_LIMIT: 4G
+
+        # Start mongo container and app before starting
+        run: |
+          cp docker-compose.mongo.yaml docker-compose.yaml
+          docker-compose up --build -d
+          npm run start &
+
+      - name: Fetch Bearer Token
+        run: |
+          ACCESS_TOKEN=$(curl --retry 30 --retry-connrefused --retry-delay 20 -d 'username=ingestor' -d 'password=aman' http://localhost:3000/api/v3/auth/login | jq -r ".access_token")
+
+      # Test the generated OpenAPI definition
+      # https://github.com/marketplace/actions/schemathesis
+      - name: Run schemathesis against the backend
+        uses: schemathesis/action@v1
+        with:
+          schema: 'http://127.0.0.1:3000/explorer-json'
+          wait-for-schema: '60'  # max wait time in seconds
+          args: '-H "Authorization: Bearer ${{ env.ACCESS_TOKEN }}"'
```

Comment:
some problems:

1.  authentication is not right
    
2. please disable the upload to schemathesis:
    https://github.com/SciCatProject/scicat-backend-next/actions/runs/7623826706/job/20764677373#step:7:200

3. Only run tests for configured endpoints (e.g. auth should only use auth/login, while  ldap, msad, oidc are not available in the test instance)
4. empty strings are not good input


My suggestion: exclude the schema test from this PR, it could come later once you have a non-failing setup.

Reply:
Using Schemathesis was mostly to try out, if we can get tests running quickly.

To answer some of your points @bpedersen2:

1. What is the problem with the authentication?
3. Sadly one can not exclude endpoints but I can explicitly define which endpoints (or endpoint patterns) to test.
4. Property testing and that is what Schemathesis is doing, does not test _good_ but _valid_ input, that's the point. An API should be able to handle valid input.

How should I proceed from here on @nitrosx? If I drop the test, this PR still includes the generation of the clients. Generating clients without testing is not the wisest choice.

Reply:
It was decided to write tests for one language.